### PR TITLE
Update ibc-go to v4

### PR DESCRIPTION
**Pull request by ulbqb**
## Description

This PR updates ibc-go to v4 along with wasmd.

According to [v3 to v4 migration guide](https://github.com/cosmos/ibc-go/blob/main/docs/migrations/v3-to-v4.md), there is nothing to handle.
- ICS27 - Interchain Accounts - Finschia doesn't have ICA controller.
- ICS29 - Fee Middleware - Finschia doesn't have 29-fee.
- Migration to fix support for base denoms with slashes - This is needed only when updating ibc-go from v3.1.0 or below to over. See [support denoms with slashes](https://github.com/cosmos/ibc-go/blob/main/docs/migrations/support-denoms-with-slashes.md).

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: scripts/generate-docs.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ fi
 # Get the path of the finschia-sdk repo and wasmd and ibc-go from go/pkg/mod
 finschia_sdk_dir=$(go list -f '{{ .Dir }}' -m github.com/Finschia/finschia-sdk) || { echo "Error: Failed to find github.com/Finschia/finschia-sdk"; exit 1; }
 wasmd_dir=$(go list -f '{{ .Dir }}' -m github.com/Finschia/wasmd) || { echo "Error: Failed to find github.com/Finschia/wasmd"; exit 1; }
-ibc_dir=$(go list -f '{{ .Dir }}' -m github.com/Finschia/ibc-go/v3) || { echo "Error: Failed to find github.com/Finschia/ibc-go/v3"; exit 1; }
+ibc_dir=$(go list -f '{{ .Dir }}' -m github.com/cosmos/ibc-go/v4) || { echo "Error: Failed to find github.com/cosmos/ibc-go/v4"; exit 1; }
```
Comment (dudong2) [inline scripts/generate-docs.sh:21]: The import is changed to cosmos/ibc-go, but the repository actually imported by replace is `finschia/ibc-go`. 
 Shouldn't we use finschia/ibc-go here?

Comment (ulbqb) [reply scripts/generate-docs.sh:21]: I confirmed that ibc-go module is in vendor/github.com/cosmos not vendor/github.com/Finschia.

**Review by dudong2 — APPROVED** [review]

File: app/app.go [hunk 2]
```diff
@@ -428,7 +428,7 @@ func NewLinkApp(
 	ibcRouter := porttypes.NewRouter()
 	ibcRouter.AddRoute(icahosttypes.SubModuleName, icaHostIBCModule).
 		AddRoute(ibctransfertypes.ModuleName, transferIBCModule).
-		AddRoute(wasmplustypes.ModuleName, wasm.NewIBCHandler(app.WasmKeeper, app.IBCKeeper.ChannelKeeper))
+		AddRoute(wasmplustypes.ModuleName, wasm.NewIBCHandler(app.WasmKeeper, app.IBCKeeper.ChannelKeeper, app.IBCKeeper.ChannelKeeper))
```
Comment (jaeseung-bae) [inline app/app.go:431]: I just wonder, is it okay to pass `app.IBCKeeper.ChannelKeeper` as the third parameter where `appVersionGetter` is expected. `ChannelKeeper` does implements `appVersionGetter` interface though. Just curious.

**Review by zemyblue — APPROVED** [review]

File: app/app.go (hunk 2 above)
Comment (ulbqb) [reply app/app.go:431]: Please check [appVersionGetter interface](https://github.com/Finschia/wasmd/blob/dedcd9ec5ba017f8a3591534ec099bc9097adb27/x/wasm/ibc.go#L21-L24) and [ChannelKeeper interface](https://github.com/Finschia/ibc-go/blob/7176e4ec5b162fe5f1a335c56e4b7b201c202d1f/modules/core/04-channel/keeper/keeper.go#L90).

**Pull request merged**